### PR TITLE
Multi-Build networking & game support

### PR DESCRIPTION
**Pull request by Kikorono**
WIP - drafting a PR for initial reviews and comments as it will be ready to merge once NPC sync is complete :) 

Multi-Build support for:
* Cache loading & parsing
* JAGGRAB/Web servers
* On-demand server
* Login
* Inbound & outbound packets
* Player & NPC sync

### CODE REVIEW

**Review by boogie-nights — COMMENTED** [review]
in the npc-sync and player-sync files I feel like we should nest the specific version implementation into like like an impl folder. Feels like it could get unwieldy without that.

**Comment by Kikorono** [comment]
> in the npc-sync and player-sync files I feel like we should nest the specific version implementation into like like an impl folder. Feels like it could get unwieldy without that.

Moved to a `builds/` dir!

File: src/net/packets/outbound-packets/sidebar-widget-packet.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { ByteBuffer } from '@runejs/common';
+import { OutboundPacket } from '../packets';
+
+interface SidebarWidgetData {
+    widgetId: number;
+    sidebarId: number;
+}
+
+export const sidebarWidgetPacket: OutboundPacket<SidebarWidgetData> = {
```
Comment (boogie-nights) [inline src/net/packets/outbound-packets/sidebar-widget-packet.ts:9]: Should we move the endcoders into an encoders into an encoders folder?

Comment (Kikorono) [reply src/net/packets/outbound-packets/sidebar-widget-packet.ts:9]: I wasn't sure if they'd get large enough, but I suppose we can if y'all wanna

Comment (boogie-nights) [reply src/net/packets/outbound-packets/sidebar-widget-packet.ts:9]: I think it would look cleaner as we add more.

File: src/net/connection.ts [hunk 2]
```diff
@@ -75,7 +75,7 @@ const handleInboundPacketData = (
 
         packet.opcode = packet.buffer.get('byte', 'u');
         packet.opcode = (packet.opcode - inCipher.rand()) & 0xff;
-        packet.size = INBOUND_PACKET_SIZES[String(packet.opcode)] ?? -3;
+        packet.size = INBOUND_PACKET_SIZES[String(player.client.connection.buildNumber)]?.[String(packet.opcode)] ?? -3;
```
Comment (boogie-nights) [inline src/net/connection.ts:78]: Should change the packet sizes object to a map. Let's open an issue for this.

**Pull request merged**